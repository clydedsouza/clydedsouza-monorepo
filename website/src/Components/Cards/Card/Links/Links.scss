.card-links {
  margin: 12px 9px;
  text-align: right;
  padding-left: 0;

  .made-using ~ & {
    margin-top: 0;
  }

  a {
    text-decoration: none;

    &:hover,
    &:focus {
      border-bottom: none;
    }

    &:last-child {
      margin-left: 9px;
    }
  }

  svg {
    width: 24px;
    margin-left: 9px;
    fill: white;
    transition: all 0.2s linear 0.2s;

    &:hover,
    &:focus {
      opacity: 0.8;
    }
  }
}
