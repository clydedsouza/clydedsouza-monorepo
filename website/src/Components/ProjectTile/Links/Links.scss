.project-links {
  margin: 12px 9px;
  text-align: right;
  padding-left: 0;

  a {
    text-decoration: none;

    &:hover,
    &:focus {
      border-bottom: none;
    }

    &:last-child {
      margin-left: 9px;
    }
  }

  i.fas,
  i.fab {
    font-size: 24px;
    margin-left: 9px;
    color: white;
    transition: all 0.2s linear 0.2s;

    &:hover,
    &:focus {
      opacity: 0.8;
    }
  }
}
