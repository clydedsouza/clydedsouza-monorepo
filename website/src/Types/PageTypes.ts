export enum PageTypes {
  About = 'About',
  Projects = 'Projects',
  Speaking = 'Speaking',
  Teaching = 'Teaching',
  Books = 'Books',
  Highlights = 'Highlights',
  Pinned = 'Pinned',
}
