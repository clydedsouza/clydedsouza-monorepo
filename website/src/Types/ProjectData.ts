export interface IProjectData {
  data: IProject[]
}

export interface IProject {
  id: string
  title: string
  date: string
  description: string
  hasWebsite: boolean
  website: string
  onGithub: boolean
  github: string
  madeUsing: string[]
  category: string
  image: string
  imageDescription: string
}
