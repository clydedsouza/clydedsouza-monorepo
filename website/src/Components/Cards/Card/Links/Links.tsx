import { sendLinkClickedEvent } from '../../../../Api/Analytics'
import { AnalyticsLinkType } from '../../../../Api/IAnalytics'
import { IProject } from '../../../../Api/IProjectData'
import GitHub from '../../../Icons/GitHub'
import Globe from '../../../Icons/Globe'
import './Links.scss'

function Links(props: Partial<IProject>) {
  if (!props.website && !props.github) return <></>

  return (
    <div className="card-links">
      {props.website && (
        <a
          href={props.website}
          target="_blank"
          rel="noreferrer"
          onClick={() =>
            sendLinkClickedEvent({
              link: props.website,
              type: AnalyticsLinkType.CardWebsite,
            })
          }
        >
          <Globe />
        </a>
      )}
      {props.github && (
        <a
          href={props.github}
          target="_blank"
          rel="noreferrer"
          onClick={() =>
            sendLinkClickedEvent({
              link: props.github,
              type: AnalyticsLinkType.CardGitHub,
            })
          }
        >
          <GitHub />
        </a>
      )}
    </div>
  )
}

export default Links
