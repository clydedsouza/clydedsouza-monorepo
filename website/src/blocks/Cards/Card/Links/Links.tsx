import { sendLinkClickedEvent } from "@/api/Analytics";
import { AnalyticsLinkType } from "@/api/IAnalytics";
import { IProject } from "@/api/IProjectData";
import GitHub from "../../../Icons/GitHub";
import Globe from "../../../Icons/Globe";
import "./Links.scss";

function Links(props: Partial<IProject>) {
  if (!props.website && !props.github) return <></>;

  return (
    <div className="card-links">
      {props.website && (
        <a
          href={props.website}
          target="_blank"
          rel="noreferrer"
          onClick={() =>
            sendLinkClickedEvent({
              link: props.website,
              type: AnalyticsLinkType.CardWebsite,
            })
          }
        >
          <Globe />
        </a>
      )}
      {props.github && (
        <a
          href={props.github}
          target="_blank"
          rel="noreferrer"
          onClick={() =>
            sendLinkClickedEvent({
              link: props.github,
              type: AnalyticsLinkType.CardGitHub,
            })
          }
        >
          <GitHub />
        </a>
      )}
    </div>
  );
}

export default Links;
