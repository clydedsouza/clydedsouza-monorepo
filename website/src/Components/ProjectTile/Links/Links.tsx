import { IProject } from '../../../Types/ProjectData'
import './Links.scss'

function Links(props: IProject) {
  return (
    <div className="project-links">
      {props.hasWebsite && (
        <a href={props.website} target="_blank" rel="noreferrer">
          <i className="fas fa-globe"></i>
        </a>
      )}
      {props.onGithub && (
        <a href={props.github} target="_blank" rel="noreferrer">
          <i className="fab fa-github"></i>
        </a>
      )}
    </div>
  )
}

export default Links
