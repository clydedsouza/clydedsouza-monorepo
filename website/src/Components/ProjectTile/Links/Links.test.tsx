import { render } from '@testing-library/react'
import { IProject } from '../../../Types/ProjectData'
import Links from './Links'

const emptyProject: IProject = {
  title: '',
  description: '',
  image: '',
  imageDescription: '',
  category: '',
  date: '',
  id: '',
  madeUsing: [],
  hasWebsite: true,
  website: 'test',
  onGithub: true,
  github: 'test',
}

describe('Links', () => {
  it.each([
    {
      hasWebsite: true,
      website: 'https://www.google.com/',
      onGithub: true,
      github: 'https://github.com/',
    },
    {
      hasWebsite: false,
      website: '',
      onGithub: true,
      github: 'https://github.com/',
    },
    {
      hasWebsite: true,
      website: 'https://www.google.com/',
      onGithub: false,
      github: '',
    },
    {
      hasWebsite: false,
      website: '',
      onGithub: false,
      github: '',
    },
  ])(
    'should render links component when hasWebsite is $hasWebsite and onGithub is $onGithub',
    ({ hasWebsite, website, onGithub, github }) => {
      const project: IProject = {
        ...emptyProject,
        hasWebsite,
        website,
        onGithub,
        github,
      }
      render(<Links {...project} />)
      expect(document.body).toMatchSnapshot()
    }
  )
})
