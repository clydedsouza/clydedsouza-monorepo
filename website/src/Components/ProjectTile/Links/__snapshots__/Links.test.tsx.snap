// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Links should render links component when hasWebsite is false and onGithub is false 1`] = `
<body>
  <div />
</body>
`;

exports[`Links should render links component when hasWebsite is false and onGithub is true 1`] = `
<body>
  <div>
    <div
      class="project-links"
    >
      <a
        href="https://github.com/"
        rel="noreferrer"
        target="_blank"
      >
        <i
          class="fab fa-github"
        />
      </a>
    </div>
  </div>
</body>
`;

exports[`Links should render links component when hasWebsite is true and onGithub is false 1`] = `
<body>
  <div>
    <div
      class="project-links"
    >
      <a
        href="https://www.google.com/"
        rel="noreferrer"
        target="_blank"
      >
        <i
          class="fas fa-globe"
        />
      </a>
    </div>
  </div>
</body>
`;

exports[`Links should render links component when hasWebsite is true and onGithub is true 1`] = `
<body>
  <div>
    <div
      class="project-links"
    >
      <a
        href="https://www.google.com/"
        rel="noreferrer"
        target="_blank"
      >
        <i
          class="fas fa-globe"
        />
      </a>
      <a
        href="https://github.com/"
        rel="noreferrer"
        target="_blank"
      >
        <i
          class="fab fa-github"
        />
      </a>
    </div>
  </div>
</body>
`;
